vignettes/adrs.Rmd

```diff
@@ -670,41 +730,128 @@ dataset_vignette(
+      event(
+        dataset_name = "ovr",
+        condition = AVALC == "PD" & ADT > RANDDT + 42,
+        set_values_to = exprs(AVALC = "Y")
+      ),
+      no_data_n
+    ),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    ignore_event_order = TRUE,
+    set_values_to = exprs(
+      PARAMCD = "ACCB",
+      PARAM = "Alternative Confirmed Clinical Benefit by Investigator",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "Recist 1.1",
+      AVAL = yn_to_numeric(AVALC),
+      ANL01FL = "Y"
+    )
+  )
+```
+
+### Considering Non-Standard Response Values
+
+Assume no evidence of disease (NED) is a valid value collected for overall
+response. A new event (`bor_ned`) can be defined for this response value and be
+added to the list of events (`events`) in the `derive_extreme_event()` call.
+
+```{r}
+bor_ned <- event(
+  description = paste(
+    "Define no evidence of disease (NED) for best overall response (BOR) as NED",
+    "occuring at least 42 days after randomization"
+  ),
+  dataset_name = "ovr",
+  condition = AVALC == "NED" & ADT >= RANDDT + 42,
+  set_values_to = exprs(AVALC = "NED")
+)
 
-All of the above steps can be repeated for different sets of records,
-such as now using assessments from the IRF/BICR
-instead of investigator. For this you would just need to replace the
-first steps with selecting the required records, and then feed these as
-input to the downstream parameter functions.
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(bor_cr, bor_pr, bor_sd, bor_non_crpd, bor_ned, bor_pd, bor_ne, no_data_missing),
+    set_values_to = exprs(
+      PARAMCD = "A1BOR",
+      PARAM = paste(
+        "Best Overall Response by Investigator (confirmation not required)",
+        "- Recist 1.1 adjusted for NED at Baseline"
+      ),
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "Recist 1.1 adjusted for NED at Baseline",
+      AVAL = aval_resp(AVALC),
+      ANL01FL = "Y"
+    )
+  )
+```
+
+## Derive Parameters using Independent Review Facility (IRF)/Blinded Independent Central Review (BICR) responses {#irf}
 
-Remember that a new progressive disease and response source object would
-be required for passing to `source_pd` and `source_resp` respectively.
+All of the above steps can be repeated for different sets of records, such as
+now using assessments from the IRF/BICR instead of investigator. For this you
+would just need to replace the first steps with selecting the required records,
+create the variables `AVALC`, `AVAL`, `ADT`, `AVISIT`, `ANL01FL`, `ANL02FL` and
+the dataset `ovrb` (see [Pre-processing of Input Records](#input)) and then feed
+these as input to the downstream parameter functions.
 
 ```{r}
-adrsirf <- rs %>%
+adrs_bicr <- rs %>%
```

Comment:
if updating this best to also update in adrs_basic and templates for consistency

Reply:
I have updated adrs_basic. In the templates we do not have this code.